### DISCUSSION TITLE
Right click shows browser context menu

### DISCUSSION THREAD

**Discussion by cgarrovillo** [visgl/deck.gl#6103] [Q&A] [answered]
Even when using Mapbox as a controller, right-clicking displays the browser context menu instead of being able to adjust the camera like the react-map-gl controls. 

Based on this doc: https://deck.gl/docs/api-reference/mapbox/overview#using-with-react
and this: https://deck.gl/docs/api-reference/react/deckgl#contextprovider

```jsx
import { MapContext, MapController } from 'react-map-gl'
const mapController = new MapController()
// ...

<DeckGL
      ref={deckRef}
      initialViewState={INITIAL_VIEW_STATE}
      onWebGLInitialized={setGLContext}
      layers={layers}
      glOptions={{
        /* To render vector tile polygons correctly */
        stencil: true,
      }}
      ContextProvider={MapContext.Provider}
      controller={mapController}>
      {glContext && (
        <>
          {/* This is important: Mapbox must be instantiated after the WebGLContext is available */}
          <ReactMapGL
            ref={mapRef}
            gl={glContext}
            mapStyle={MAP_STYLE}
            onLoad={onMapLoad}
            width='100%'
            height='100vh'
            {...viewport}
            onViewportChange={setViewport}
            />
        </>
      )}
</DeckGL>

```

How do I achieve the same functionality as ReactMapGL in DeckGL with React? If DeckGL's `ContextProvider` and `controller` are using context & controller from `react-map-gl`,  why is the functionality behaving differently?

**Comment by Pessimistress** [comment, marked as answer]
> If DeckGL's ContextProvider and controller are using context & controller from react-map-gl, why is the functionality behaving differently?

When you use `DeckGL` as the parent, you are using DeckGL's controller. It has identical implementation as react-map-gl's controller, but different defaults for backward-compatibility reasons.

You can disable the context menu like this:

```js
<div onContextMenu={evt => evt.preventDefault()}>
  <DeckGL ... />
</div>
```

**Reply by cgarrovillo** [reply]
Makes sense.  Curious though, it says in the docs that the `<DeckGL />` component [accepts all of Deck's properties](https://deck.gl/docs/api-reference/react/deckgl#properties),  and in the Deck API the controller can be [overridden](https://deck.gl/docs/api-reference/core/deck#interaction-settings).  In which case if I use the controller from react-map-gl, shouldn't that override DeckGL's default controller?

**Reply by Pessimistress** [reply]
If you follow the "use with React" example on the documentation page, then you are always using DeckGL's controller.

You can use react-map-gl's controller if you use `ReactMapGL` as the parent of `DeckGL`:

```js
return <ReactMapGL {...viewport} onViewportChange={setViewport}>
  <DeckGL viewState={viewport} controller={false} />
</ReactMapGL>
```

If you want to use `MapboxLayer` with mapbox's native controller, like how the pure-js example works, then you do not render a `DeckGL` component at all:

```js
const addDeckLayers = () => {
  const myDeckLayer = new MapboxLayer({
    id: 'my-scatterplot',
    type: ScatterplotLayer,
    data: [
      {position: [-74.5, 40], size: 100}
    ],
    getPosition: d => d.position,
    getRadius: d => d.size,
    getColor: [255, 0, 0]
  });
  mapRef.current.getMap().addLayer(myDeckLayer);
};

return <ReactMapGL ... ref={mapRef} onLoad={addDeckLayers} />;
```

**Reply by sp21** [reply]
Try to use 
@HostListener ("contextmenu",["$event"])
onRightClick(event:any){
  event.preventDefault();
  console.log(event.clientX);
}
and use (contextmenu) event on the template
